relay-server/src/utils/envelope_context.rs

```diff
@@ -83,6 +83,12 @@ impl EnvelopeContext {
         self
     }
 
+    /// TODO(ja): What about this?
+    pub fn tmp_set_metrics_extracted(&mut self) -> &mut Self {
+        self.summary.event_metrics_extracted = true;
+        self
+    }
```

Comment:
I am not happy with adding this method. We need it now since the `EnvelopeProcessor` drops an envelope and calls `EnvelopeContext::reject` after metrics have been extracted but before the EnvelopeContext has been updated with this flag: 
 - We need to update it as soon as metrics extraction has happened, so that the call to `reject` can use the updated summary.
- However, we also cannot simply call `EnvelopeContext::update` because the Transaction item has been removed from the Envelope during this time.

Reply:
With adding this documentation from the comment above , explaining what this is for, it make sense.  
Since this method called only in my place I wouldn't add it at all -  only just for purpose of the documentation.

Reply:
It would be nice to have an envelope summary that stays up-to-date automatically. That is, a wrapper around the envelope that caches computed properties and invalidates them when the envelope changes. Since that would be too big of a refactor for today, I think this method is fine.